subprojects/core/src/main/java/org/gradle/testfixtures/ProjectBuilder.java

```diff
@@ -70,7 +71,7 @@ public static ProjectBuilder builder() {
      * @param dir The project directory
      * @return The builder
      */
-    public ProjectBuilder withProjectDir(File dir) {
+    public ProjectBuilder withProjectDir(@Nullable File dir) {
```

Comment:
Do we really want to change the public API to make this parameter nullable? What would passing `null` mean? If we decide to go with a nullable parameter, then we should update the javadocs to mention what passing `null` does.

Reply:
As far as I understand, in this particular case this is not an API change.

The reason why I added the `@Nullable` annotation is because I also opted-in for `NonNullApi` in the package that contains `ProjectBuilder.java`. As such, making the arguments nullable is not a change, but rather the preservation of the nullability of how it was before the PR.
https://github.com/gradle/gradle/pull/21331/files/e23c89e6b94a5c729fd899f4de9f801c32e04dfc#diff-2b8f9a11d95563e1a71d2b8a216af34b5f4803aa5160e54497b08da3b322a497R20

This applies to all the arguments marked nullable in this PR. However, there is one setter `withName` that was deliberately not marked nullable, and can be considered as API change: it was nullable before, and became non-nullable after the opt-in. However, the expectation in the implementation code already was that the `name` field of `ProjectBuilder` is non-nullable.